tests/test_appelb.py

```diff
@@ -649,3 +649,99 @@ def test_rds_health_events_filter(self):
         )
         resources = policy.run()
         self.assertEqual(len(resources), 1)
+
+
+class TestModifyVpcSecurityGroupsAction(BaseTest):
+
+    def test_appelb_remove_matched_security_groups(self):
+
+        # Test conditions:
+        #   - list contains only one ALB, 'test-abc'
+        #   - has two SGs attached before, 1 SG after
+        #   - test checks name of ALB is correct and
+        #   - that SGs change and are expected values
+
+        session_factory = self.replay_flight_data(
+            "test_appelb_remove_matched_security_groups"
+        )
+
+        p = self.load_policy(
+            {
+                "name": "appelb-remove-matched-security-groups",
+                "resource": "app-elb",
+                "filters": [
+                    {
+                        "type": "value",
+                        "key": "LoadBalancerName",
+                        "value": "test-abc",
+                        "op": "eq",
+                    },
+                    {
+                        "type": "security-group",
+                        "key": "GroupName",
+                        "value": ".*controllers",
+                        "op": "regex",
+                    },
+                ],
+                "actions": [
+                    {
+                        "type": "modify-security-groups",
+                        "remove": "matched",
+                        "isolation-group": "sg-01a19f602ecaf25f4",
+                    }
+                ],
+            },
+            session_factory=session_factory,
+        )
+
+        resources = p.run()
+        client = session_factory().client('elbv2')
+        response = client.describe_load_balancers()
+        clean_resources = response['LoadBalancers']
+
+        self.assertEqual(len(resources), 1)
+        self.assertIn("test", resources[0]["LoadBalancerName"])
+        self.assertEqual(len(resources[0]["SecurityGroups"]), 2)
```

Comment:
should also assert that the security groups that were on the resource previously are the ones that were actually matched

Reply:
added "before" test as suggested, passes